GoogleForCreators/web-stories-wp
# Include border into the element selection line

## Context
Originally only an "outer" border was implemented outside of the selection line.
It was discussed later that the border should also be included in the selection line, so the user can snap by the border and also select an element by clicking on the border.


## Summary
This PR adjusts the frame, edit, display mode to include the border into the selection line.




## Relevant Technical Choices
- Note that the border width does not change the width and height of the element directly as originally intended since the border is in pixels and not responsive.
- Non-rectangular shapes have a different border logic and are already displaying the frame somewhere in the middle of the border -- these will not have the selection line perfectly including the border now either. This change mainly applies to rectangular elements / shapes.
- Edit mode stays the same for media -- border is not included in the edit mode.


## To-do

- [x] Fix display with unlocked border
- [x] Fix text edit with unlocked border
- [x] Fix resizing multiple elements
- [x] Add new test for selecting by clicking on border



## User-facing changes
![text-edit](https://user-images.githubusercontent.com/3294597/180406256-52b70a59-4acd-4c27-ab49-0c46bd4c2377.gif)
![border-resizing](https://user-images.githubusercontent.com/3294597/180406270-457b844d-f2e5-4ab4-918a-1c1cced141b1.gif)
![border-display](https://user-images.githubusercontent.com/3294597/180406277-9b6e0eca-9f53-44d3-8566-e35a92b6b758.gif)




## Testing Instructions






This PR can be tested by following these steps:

1. Add rectangular elements with border (media, text, shape)
2. Increase and decrease the border -- verify that border is now included in the selection line.
3. Verify it's possible to drag from the border
4. Verify it's possible to select an element by clicking on the border
5. Verify that resizing still works and isn't jumpy / changing the element size unexpectedly.
6. Verify that border is also included in the selection in text edit mode.
7. Verify that media edit mode does not include border (is as before)
8. Verify that output is not influenced by this change.
9. Verify that when adding border to an element, the content of the element stays in the same place.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #5325
